storage/innobase/include/sync0policy.h

```diff
@@ -221,7 +234,7 @@ class MutexDebug {
 	os_thread_id_t get_thread_id() const
 		UNIV_NOTHROW
 	{
-		return(m_context.m_thread_id);
+		return(m_context.get_thread_id());
```

Comment:
This should not be needed. As discussed it is supposed to be protected by ```synch_check_*()``` stuff. Or you disagree?

Reply:
```
WARNING: ThreadSanitizer: data race (pid=22092)
  Atomic write of size 8 at 0x7b6c00001548 by thread T8:
    #0 __tsan_atomic64_store <null> (libtsan.so.0+0x0000000634c2)
    #1 MutexDebug<TTASEventMutex<GenericPolicy> >::Context::release() storage/innobase/include/sync0policy.h:92
    #2 MutexDebug<TTASEventMutex<GenericPolicy> >::release(TTASEventMutex<GenericPolicy> const*) storage/innobase/include/sync0policy.ic:96
    #3 GenericPolicy<TTASEventMutex<GenericPolicy> >::release(TTASEventMutex<GenericPolicy> const&) storage/innobase/include/sync0policy.h:369
    #4 PolicyMutex<TTASEventMutex<GenericPolicy> >::exit() storage/innobase/include/ib0mutex.h:607
    #5 buf_page_io_complete(buf_page_t*, bool) storage/innobase/buf/buf0buf.cc:6128
    #6 fil_aio_wait(unsigned long) storage/innobase/fil/fil0fil.cc:5500
    #7 io_handler_thread storage/innobase/srv/srv0start.cc:340
    #8 <null> <null> (libtsan.so.0+0x000000025aab)

  Previous read of size 8 at 0x7b6c00001548 by thread T11 (mutexes: write M1208):
    #0 MutexDebug<TTASEventMutex<GenericPolicy> >::get_thread_id() const storage/innobase/include/sync0policy.h:224
    #1 sync_array_detect_deadlock storage/innobase/sync/sync0arr.cc:750
    #2 sync_array_wait_event(sync_array_t*, sync_cell_t*&) storage/innobase/sync/sync0arr.cc:465
    #3 TTASEventMutex<GenericPolicy>::enter(unsigned int, unsigned int, char const*, unsigned int) storage/innobase/include/ib0mutex.h:516
    #4 PolicyMutex<TTASEventMutex<GenericPolicy> >::enter(unsigned int, unsigned int, char const*, unsigned int) storage/innobase/include/ib0mutex.h:635
    #5 buf_page_io_complete(buf_page_t*, bool) storage/innobase/buf/buf0buf.cc:6049
    #6 fil_aio_wait(unsigned long) storage/innobase/fil/fil0fil.cc:5500
    #7 io_handler_thread storage/innobase/srv/srv0start.cc:340
    #8 <null> <null> (libtsan.so.0+0x000000025aab)

  Location is heap block of size 1568 at 0x7b6c00001500 allocated by main thread:
    #0 calloc <null> (libtsan.so.0+0x0000000282a6)
    #1 buf_pool_init(unsigned long, unsigned long) storage/innobase/buf/buf0buf.cc:2002
    #2 innobase_start_or_create_for_mysql() storage/innobase/srv/srv0start.cc:1825
    #3 innobase_init storage/innobase/handler/ha_innodb.cc:4234
    #4 ha_initialize_handlerton(st_plugin_int*) sql/handler.cc:521
    #5 plugin_initialize sql/sql_plugin.cc:1440
    #6 plugin_init(int*, char**, int) sql/sql_plugin.cc:1722
    #7 init_server_components sql/mysqld.cc:5321
    #8 mysqld_main(int, char**) sql/mysqld.cc:5916
    #9 main sql/main.cc:25
```